### DISCUSSION TITLE
Should the documentation for Devtools be updated?

### DISCUSSION THREAD

**Discussion by tbassetto** [TanStack/router#1016] [General]
I am looking at https://tanstack.com/router/v1/docs/devtools and it is using `process.env.NODE_ENV` which won't work in Vite (which is the recommended bundler in the rest of the doc).

As soon as I changed this snippet to use `import.env.NODE_ENV`, I got this error in my app:
<img width="1093" alt="Screenshot 2024-01-18 at 13 58 27" src="https://github.com/TanStack/router/assets/211630/ad84c93d-8a21-40c6-a327-5baf89fd1a3b">

I understand that now the entire app should be wrapped in a `<Suspense>...</Suspense>` because of the lazy loading of the devtools. I think this should be mentioned too in the documentation, provided that I got it right :)

**Comment by SeanCassiere** [comment]
It doesn't have to be that specific environment variable.

I've seen projects use a variable like `VITE_APP_DEPLOYMENT_ENV` and set values like `dev`, `qa`, `staging`, and `production`.
That could be the condition for you to check before importing.
